trainer/preprocessing.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2020 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Preprocessing and Infeeding Tools For Redial and MovieLens Data."""
+
+import tensorflow.compat.v1 as tf
+import tensorflow_datasets as tfds
+import functools
+import os
+import json
+
+BASE_DIR = "gs://e2e_central"
+DATA_DIR = os.path.join(BASE_DIR, "data")
+
+rd_tsv_path = {
+    "train": os.path.join(DATA_DIR, "rd-train.tsv"),
+    "validation": os.path.join(DATA_DIR, "rd-validation.tsv")
+}
+
+def rd_jsonl_to_tsv(in_fname, out_fname):
+  """Converts the redial jsonl to a tsv."""
+  print("Reading: " + in_fname)
+  def fix_spacing(text):
+    """Removes extra spaces."""
+    # Remove incorrect spacing around punctuation.
+    text = text.replace(" ,", ",").replace(" .", ".").replace(" %", "%")
+    text = text.replace(" - ", "-").replace(" : ", ":").replace(" / ", "/")
+    text = text.replace("( ", "(").replace(" )", ")")
+    text = text.replace("`` ", "\"").replace(" ''", "\"")
+    text = text.replace(" 's", "'s").replace("s ' ", "s' ")
+    return text
+
+  count = 0
+  with tf.io.gfile.GFile(in_fname, "rb") as infile,\
+      tf.io.gfile.GFile(out_fname, "w") as outfile:
+    for line in infile:
+      ex = json.loads(line)
+      print(line)
```

Comment:
Is this print a debug-only kind of print?

Reply:
yup, got rid of it